visionmetrics/detection/confusion_matrix.py

```diff
@@ -0,0 +1,171 @@
+import torch
+from torchmetrics import Metric
+
+
+class DetectionConfusionMatrix(Metric):
+    """
+        Calcuates confusion matrix in the context of object detection.
+
+        Parameters
+        ----------
+        predictions : list of lists
+            Each list contains list of bounding box predictions for an image in the format [class_id, score, xmin, ymin, xmax, ymax]
+        targets : list of lists
+            Each list contains ground truth bounding boxes for an image in the format [class_id, xmin, ymin, xmax, ymax]
+        iou_threshold : float
+            IoU threshold to consider a detection as True Positive
+
+        Example:
+        predictions = [[[0, 0.9, 10, 20, 50, 100], [1, 0.8, 30, 40, 80, 120]], [[1, 0.7, 20, 30, 60, 90]]]
+        targets = [[[0, 10, 20, 50, 100], [1, 30, 40, 80, 120]], [[1, 20, 30, 60, 90]]]
+
+        Returns
+        -------
+        confusion_matrix : dict
+            Dictionary containing the counts of TP, FP, and FN, as well as details for FP reasons
+
+            TP: When a predicted bounding box has IoU greater than the threshold with a ground truth box of the same class
+            FN: When a ground truth bounding box has no corresponding predicted bounding box
+            FP:
+                1. FP_due_to_wrong_class: IOU >= threshold but predicted class is different from the ground truth class.
+                2. FP_due_to_low_iou: Predicted bbox IOU < threshold (including no overlap)
+                3. fp_due_to_extra_pred_boxes: Excess predicted bboxes when all ground truth boxes have been matched
+
+        """
+
+    def __init__(self, iou_threshold=0.5, **kwargs):
```

Comment:
`predictions` and `targets` are added using the update() method in the Metric class.

Reply:
So `predictions` and `targets` are not part of the `kwargs` here? So why are they on the `__init__` docstring?

Reply:
I just intended to provide an overall info on what the class does. I agree this could be confusing for first time user. 
I rephrased the class docstring for more clarity